### ISSUE TITLE
routing -= (1. - presence) * 1e32

### ISSUE THREAD

**Issue by lithostark** [akosiorek/stacked_capsule_autoencoders#3]
Hi, it is a great model and thank you for sharing the code. I have one question about how the attention is calculated in "QKVAttention". I'm having difficulty to understand how multiplying such a large number, 1e32, to the "presence" of part capsules would help with the "routing". I think the variants among the presence would overwhelm the difference in routing. Could you explain in detail please? Thank you!

**Comment by akosiorek** [comment]
This op is performed in logit space (inputs to softmax), and effectively subtracts 1e32 from the logits for a given input. This is means that attending to this input is not possible, and behaves essentially the same as not providing this input.
